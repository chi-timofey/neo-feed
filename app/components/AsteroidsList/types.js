import { string, number, shape, arrayOf } from 'prop-types';

export const AsteroidItemType = shape({
  estimated_diameter: shape({
    kilometers: shape({
      estimated_diameter_max: number.isRequired,
    }),
  }),
  absolute_magnitude_h: number.isRequired,
  close_approach_data: arrayOf(shape({
    miss_distance: shape({
      kilometers: string.isRequired,
    }),
  })),
  name: string.isRequired,
});
