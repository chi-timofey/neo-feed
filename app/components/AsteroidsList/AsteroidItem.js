/* eslint-disable camelcase */
import React, { PureComponent } from 'react';
import classnames from 'classnames';
import { bool } from 'prop-types';
import { AsteroidItemType } from './types';

class AsteroidItem extends PureComponent {
  static propTypes = {
    item: AsteroidItemType.isRequired,
    highlighted: bool.isRequired,
  }

  render() {
    const { item, highlighted } = this.props;
    const {
      estimated_diameter, absolute_magnitude_h, close_approach_data, name,
    } = item;
    return (
      <tr className={classnames({ 'table-danger': highlighted })}>
        <th scope="row">{name}</th>
        <td>{estimated_diameter.kilometers.estimated_diameter_max}</td>
        <td>{absolute_magnitude_h}</td>
        <td>{close_approach_data[0].miss_distance.kilometers}</td>
      </tr>
    );
  }
}

export default AsteroidItem;
