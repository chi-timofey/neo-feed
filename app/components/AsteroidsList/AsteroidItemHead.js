import React from 'react';

const AsteroidItemHead = () => (
  <thead className="thead-light">
    <tr>
      <th scope="col">Name</th>
      <th scope="col">Estimated Diameter</th>
      <th scope="col">Absolute Magnitude_h</th>
      <th scope="col">Close Approach Data</th>
    </tr>
  </thead>
);

export default AsteroidItemHead;
