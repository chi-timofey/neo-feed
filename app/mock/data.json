[
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/2153460?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "2153460",
        "name": "153460 (2001 RN)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=2153460",
        "absolute_magnitude_h": 19.6,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.3195618867,
                "estimated_diameter_max": 0.7145621017
            },
            "meters": {
                "estimated_diameter_min": 319.5618867213,
                "estimated_diameter_max": 714.5621017269
            },
            "miles": {
                "estimated_diameter_min": 0.1985664891,
                "estimated_diameter_max": 0.4440081677
            },
            "feet": {
                "estimated_diameter_min": 1048.4314204307,
                "estimated_diameter_max": 2344.3639258298
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "14.4616613971",
                    "kilometers_per_hour": "52061.9810294783",
                    "miles_per_hour": "32349.2903791222"
                },
                "miss_distance": {
                    "astronomical": "0.3000236917",
                    "lunar": "116.7092208862",
                    "kilometers": "44882908",
                    "miles": "27888946"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/2446833?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "2446833",
        "name": "446833 (2001 RB12)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=2446833",
        "absolute_magnitude_h": 21,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.1677084622,
                "estimated_diameter_max": 0.3750075218
            },
            "meters": {
                "estimated_diameter_min": 167.7084621628,
                "estimated_diameter_max": 375.0075217981
            },
            "miles": {
                "estimated_diameter_min": 0.1042091748,
                "estimated_diameter_max": 0.2330187988
            },
            "feet": {
                "estimated_diameter_min": 550.2246310023,
                "estimated_diameter_max": 1230.3396778159
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "8.3570541243",
                    "kilometers_per_hour": "30085.3948473585",
                    "miles_per_hour": "18693.8943705712"
                },
                "miss_distance": {
                    "astronomical": "0.2274034746",
                    "lunar": "88.4599533081",
                    "kilometers": "34019076",
                    "miles": "21138474"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3117459?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3117459",
        "name": "(2002 FQ5)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3117459",
        "absolute_magnitude_h": 20.5,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.2111324448,
                "estimated_diameter_max": 0.4721064988
            },
            "meters": {
                "estimated_diameter_min": 211.1324447897,
                "estimated_diameter_max": 472.1064988055
            },
            "miles": {
                "estimated_diameter_min": 0.1311915784,
                "estimated_diameter_max": 0.2933532873
            },
            "feet": {
                "estimated_diameter_min": 692.6917701639,
                "estimated_diameter_max": 1548.9058855411
            }
        },
        "is_potentially_hazardous_asteroid": true,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "29.4014251824",
                    "kilometers_per_hour": "105845.130656802",
                    "miles_per_hour": "65768.0479906112"
                },
                "miss_distance": {
                    "astronomical": "0.4714403955",
                    "lunar": "183.3903198242",
                    "kilometers": "70526480",
                    "miles": "43823124"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3472698?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3472698",
        "name": "(2009 UX87)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3472698",
        "absolute_magnitude_h": 25.3,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.0231502122,
                "estimated_diameter_max": 0.0517654482
            },
            "meters": {
                "estimated_diameter_min": 23.150212221,
                "estimated_diameter_max": 51.7654482198
            },
            "miles": {
                "estimated_diameter_min": 0.0143848705,
                "estimated_diameter_max": 0.0321655483
            },
            "feet": {
                "estimated_diameter_min": 75.9521422633,
                "estimated_diameter_max": 169.8341531374
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "9.0878017789",
                    "kilometers_per_hour": "32716.0864041408",
                    "miles_per_hour": "20328.503799284"
                },
                "miss_distance": {
                    "astronomical": "0.2224939584",
                    "lunar": "86.5501480103",
                    "kilometers": "33284624",
                    "miles": "20682106"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3709081?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3709081",
        "name": "(2015 BW510)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3709081",
        "absolute_magnitude_h": 20.7,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.1925550782,
                "estimated_diameter_max": 0.4305662442
            },
            "meters": {
                "estimated_diameter_min": 192.5550781879,
                "estimated_diameter_max": 430.566244241
            },
            "miles": {
                "estimated_diameter_min": 0.1196481415,
                "estimated_diameter_max": 0.2675413778
            },
            "feet": {
                "estimated_diameter_min": 631.7424027221,
                "estimated_diameter_max": 1412.6189567557
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "13.5198173946",
                    "kilometers_per_hour": "48671.3426207033",
                    "miles_per_hour": "30242.4795300698"
                },
                "miss_distance": {
                    "astronomical": "0.4779279498",
                    "lunar": "185.9139709473",
                    "kilometers": "71497008",
                    "miles": "44426180"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3752432?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3752432",
        "name": "(2016 HE3)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3752432",
        "absolute_magnitude_h": 24.3,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.0366906138,
                "estimated_diameter_max": 0.0820427065
            },
            "meters": {
                "estimated_diameter_min": 36.6906137531,
                "estimated_diameter_max": 82.0427064882
            },
            "miles": {
                "estimated_diameter_min": 0.0227984834,
                "estimated_diameter_max": 0.0509789586
            },
            "feet": {
                "estimated_diameter_min": 120.3760332259,
                "estimated_diameter_max": 269.1689931548
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "10.7253847904",
                    "kilometers_per_hour": "38611.3852453262",
                    "miles_per_hour": "23991.6132375751"
                },
                "miss_distance": {
                    "astronomical": "0.1161466413",
                    "lunar": "45.1810417175",
                    "kilometers": "17375290",
                    "miles": "10796505"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3769888?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3769888",
        "name": "(2017 DC36)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3769888",
        "absolute_magnitude_h": 22.1,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.1010543415,
                "estimated_diameter_max": 0.2259643771
            },
            "meters": {
                "estimated_diameter_min": 101.054341542,
                "estimated_diameter_max": 225.9643771094
            },
            "miles": {
                "estimated_diameter_min": 0.0627922373,
                "estimated_diameter_max": 0.140407711
            },
            "feet": {
                "estimated_diameter_min": 331.5431259047,
                "estimated_diameter_max": 741.3529669956
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "6.3425658805",
                    "kilometers_per_hour": "22833.2371699691",
                    "miles_per_hour": "14187.6856181955"
                },
                "miss_distance": {
                    "astronomical": "0.2080938155",
                    "lunar": "80.9484939575",
                    "kilometers": "31130392",
                    "miles": "19343530"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3771633?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3771633",
        "name": "(2017 FZ2)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3771633",
        "absolute_magnitude_h": 26.7,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.0121494041,
                "estimated_diameter_max": 0.0271668934
            },
            "meters": {
                "estimated_diameter_min": 12.14940408,
                "estimated_diameter_max": 27.1668934089
            },
            "miles": {
                "estimated_diameter_min": 0.0075492874,
                "estimated_diameter_max": 0.0168807197
            },
            "feet": {
                "estimated_diameter_min": 39.8602508817,
                "estimated_diameter_max": 89.1302305717
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "6.4171733855",
                    "kilometers_per_hour": "23101.8241876675",
                    "miles_per_hour": "14354.5751459426"
                },
                "miss_distance": {
                    "astronomical": "0.0708396096",
                    "lunar": "27.5566062927",
                    "kilometers": "10597454",
                    "miles": "6584953"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3772993?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3772993",
        "name": "(2017 FY101)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3772993",
        "absolute_magnitude_h": 27.2,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.0096506147,
                "estimated_diameter_max": 0.0215794305
            },
            "meters": {
                "estimated_diameter_min": 9.6506146958,
                "estimated_diameter_max": 21.5794304844
            },
            "miles": {
                "estimated_diameter_min": 0.0059966121,
                "estimated_diameter_max": 0.0134088323
            },
            "feet": {
                "estimated_diameter_min": 31.6621227185,
                "estimated_diameter_max": 70.7986587106
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "14.8231783796",
                    "kilometers_per_hour": "53363.442166434",
                    "miles_per_hour": "33157.967717249"
                },
                "miss_distance": {
                    "astronomical": "0.1272148636",
                    "lunar": "49.4865837097",
                    "kilometers": "19031072",
                    "miles": "11825360"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3675307?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3675307",
        "name": "(2014 NZ64)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3675307",
        "absolute_magnitude_h": 22.9,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.0699125232,
                "estimated_diameter_max": 0.1563291544
            },
            "meters": {
                "estimated_diameter_min": 69.9125232246,
                "estimated_diameter_max": 156.3291544087
            },
            "miles": {
                "estimated_diameter_min": 0.0434416145,
                "estimated_diameter_max": 0.097138403
            },
            "feet": {
                "estimated_diameter_min": 229.3718026961,
                "estimated_diameter_max": 512.8909429502
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "15.5290368098",
                    "kilometers_per_hour": "55904.5325151378",
                    "miles_per_hour": "34736.9024397533"
                },
                "miss_distance": {
                    "astronomical": "0.2512262445",
                    "lunar": "97.7270126343",
                    "kilometers": "37582912",
                    "miles": "23352938"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3780765?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3780765",
        "name": "(2017 QB2)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3780765",
        "absolute_magnitude_h": 24.4,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.0350392641,
                "estimated_diameter_max": 0.0783501764
            },
            "meters": {
                "estimated_diameter_min": 35.0392641108,
                "estimated_diameter_max": 78.3501764334
            },
            "miles": {
                "estimated_diameter_min": 0.0217723826,
                "estimated_diameter_max": 0.0486845275
            },
            "feet": {
                "estimated_diameter_min": 114.9582192654,
                "estimated_diameter_max": 257.0543928497
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "6.8976797893",
                    "kilometers_per_hour": "24831.6472415022",
                    "miles_per_hour": "15429.419920699"
                },
                "miss_distance": {
                    "astronomical": "0.2046959492",
                    "lunar": "79.6267242432",
                    "kilometers": "30622080",
                    "miles": "19027678"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3781917?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3781917",
        "name": "(2017 SV17)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3781917",
        "absolute_magnitude_h": 25.7,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.0192555078,
                "estimated_diameter_max": 0.0430566244
            },
            "meters": {
                "estimated_diameter_min": 19.2555078188,
                "estimated_diameter_max": 43.0566244241
            },
            "miles": {
                "estimated_diameter_min": 0.0119648141,
                "estimated_diameter_max": 0.0267541378
            },
            "feet": {
                "estimated_diameter_min": 63.1742402722,
                "estimated_diameter_max": 141.2618956756
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "10.2221721464",
                    "kilometers_per_hour": "36799.8197271203",
                    "miles_per_hour": "22865.976874332"
                },
                "miss_distance": {
                    "astronomical": "0.3128367871",
                    "lunar": "121.6935119629",
                    "kilometers": "46799720",
                    "miles": "29079998"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3802036?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3802036",
        "name": "(2018 FB3)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3802036",
        "absolute_magnitude_h": 25.187,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.024386809,
                "estimated_diameter_max": 0.0545305626
            },
            "meters": {
                "estimated_diameter_min": 24.3868089865,
                "estimated_diameter_max": 54.5305626481
            },
            "miles": {
                "estimated_diameter_min": 0.0151532559,
                "estimated_diameter_max": 0.0338837102
            },
            "feet": {
                "estimated_diameter_min": 80.0092183953,
                "estimated_diameter_max": 178.9060511585
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "15.8632847344",
                    "kilometers_per_hour": "57107.825043777",
                    "miles_per_hour": "35484.5816223402"
                },
                "miss_distance": {
                    "astronomical": "0.0238236554",
                    "lunar": "9.2674026489",
                    "kilometers": "3563968.25",
                    "miles": "2214547.25"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3802039?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3802039",
        "name": "(2018 FD3)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3802039",
        "absolute_magnitude_h": 22.315,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.0915282116,
                "estimated_diameter_max": 0.204663303
            },
            "meters": {
                "estimated_diameter_min": 91.5282115969,
                "estimated_diameter_max": 204.6633029896
            },
            "miles": {
                "estimated_diameter_min": 0.0568729764,
                "estimated_diameter_max": 0.1271718412
            },
            "feet": {
                "estimated_diameter_min": 300.2894177356,
                "estimated_diameter_max": 671.4675509805
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "2.6365314494",
                    "kilometers_per_hour": "9491.5132178581",
                    "miles_per_hour": "5897.6571991741"
                },
                "miss_distance": {
                    "astronomical": "0.2694721181",
                    "lunar": "104.8246536255",
                    "kilometers": "40312456",
                    "miles": "25048998"
                },
                "orbiting_body": "Earth"
            }
        ]
    },
    {
        "links": {
            "self": "https://api.nasa.gov/neo/rest/v1/neo/3802000?api_key=SPboJP8XCDF9nlUzSqcqzh0Mq9sJuy6Hf27FuTFl"
        },
        "neo_reference_id": "3802000",
        "name": "(2018 FQ1)",
        "nasa_jpl_url": "http://ssd.jpl.nasa.gov/sbdb.cgi?sstr=3802000",
        "absolute_magnitude_h": 26.927,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 0.0109434681,
                "estimated_diameter_max": 0.0244703386
            },
            "meters": {
                "estimated_diameter_min": 10.9434681079,
                "estimated_diameter_max": 24.4703385988
            },
            "miles": {
                "estimated_diameter_min": 0.0067999537,
                "estimated_diameter_max": 0.0152051588
            },
            "feet": {
                "estimated_diameter_min": 35.903767907,
                "estimated_diameter_max": 80.2832656885
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "2018-03-25",
                "epoch_date_close_approach": 1521961200000,
                "relative_velocity": {
                    "kilometers_per_second": "5.9388604155",
                    "kilometers_per_hour": "21379.8974958714",
                    "miles_per_hour": "13284.6368634763"
                },
                "miss_distance": {
                    "astronomical": "0.0144704833",
                    "lunar": "5.6290178299",
                    "kilometers": "2164753.5",
                    "miles": "1345115.5"
                },
                "orbiting_body": "Earth"
            }
        ]
    }
]