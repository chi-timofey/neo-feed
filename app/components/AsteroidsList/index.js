/* eslint-disable camelcase */
import React from 'react';
import { string, arrayOf } from 'prop-types';
import { AsteroidItemType } from './types';
import AsteroidItem from './AsteroidItem';
import AsteroidItemHead from './AsteroidItemHead';

const AsteroidsList = ({ items, highlightedListId }) => (
  <table className="table">
    <AsteroidItemHead />
    <tbody>
      {
        items.map(item => (
          <AsteroidItem
            highlighted={highlightedListId.includes(item.neo_reference_id)}
            key={item.neo_reference_id}
            item={item}
          />))
      }
    </tbody>
  </table>
);

AsteroidsList.propTypes = {
  items: arrayOf(AsteroidItemType).isRequired,
  highlightedListId: arrayOf(string).isRequired,
};

export default AsteroidsList;
