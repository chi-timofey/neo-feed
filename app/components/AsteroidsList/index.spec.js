import React from 'react';
import renderer from 'react-test-renderer';
import data from 'mock/data.json';
import AsteroidsList from '.';

describe('<AsteroidsList />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<AsteroidsList items={data.slice(0, 6)} highlightedListId={['123', '234']} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
