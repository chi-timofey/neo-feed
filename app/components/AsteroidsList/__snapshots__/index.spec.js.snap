// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AsteroidsList /> renders correctly 1`] = `
<table
  className="table"
>
  <thead
    className="thead-light"
  >
    <tr>
      <th
        scope="col"
      >
        Name
      </th>
      <th
        scope="col"
      >
        Estimated Diameter
      </th>
      <th
        scope="col"
      >
        Absolute Magnitude_h
      </th>
      <th
        scope="col"
      >
        Close Approach Data
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className=""
    >
      <th
        scope="row"
      >
        153460 (2001 RN)
      </th>
      <td>
        0.7145621017
      </td>
      <td>
        19.6
      </td>
      <td>
        44882908
      </td>
    </tr>
    <tr
      className=""
    >
      <th
        scope="row"
      >
        446833 (2001 RB12)
      </th>
      <td>
        0.3750075218
      </td>
      <td>
        21
      </td>
      <td>
        34019076
      </td>
    </tr>
    <tr
      className=""
    >
      <th
        scope="row"
      >
        (2002 FQ5)
      </th>
      <td>
        0.4721064988
      </td>
      <td>
        20.5
      </td>
      <td>
        70526480
      </td>
    </tr>
    <tr
      className=""
    >
      <th
        scope="row"
      >
        (2009 UX87)
      </th>
      <td>
        0.0517654482
      </td>
      <td>
        25.3
      </td>
      <td>
        33284624
      </td>
    </tr>
    <tr
      className=""
    >
      <th
        scope="row"
      >
        (2015 BW510)
      </th>
      <td>
        0.4305662442
      </td>
      <td>
        20.7
      </td>
      <td>
        71497008
      </td>
    </tr>
    <tr
      className=""
    >
      <th
        scope="row"
      >
        (2016 HE3)
      </th>
      <td>
        0.0820427065
      </td>
      <td>
        24.3
      </td>
      <td>
        17375290
      </td>
    </tr>
  </tbody>
</table>
`;
